### PR TITLE
fix #15463

### LINKED ISSUE #15463: VM segmentation faults with swap
Found through trying general examples from #15461. I swear I checked if this is a duplicate, I couldn't find anything.

### Example
```nim
static:
  var s = @[1, 2, 3]
  swap(s[0], s[2])
```

### Current Output
```
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
```

### Expected Output
s becomes @[3, 2, 1]

### Additional Information

* In versions before 0.19, this says `cannot evaluate at compile time: :tmp`, and ever since 0.19 this segmentation faults.

```
$ nim -v
Nim Compiler Version 1.3.5 [Windows: amd64]
(latest devel)
```

### PR DESCRIPTION
**Pull request by ringabout**

### CODE REVIEW

File: compiler/lowerings.nim [hunk 1]
```diff
@@ -135,9 +135,14 @@ proc lowerTupleUnpackingForAsgn*(g: ModuleGraph; n: PNode; idgen: IdGenerator; o
 proc lowerSwap*(g: ModuleGraph; n: PNode; idgen: IdGenerator; owner: PSym): PNode =
   result = newNodeI(nkStmtList, n.info)
   # note: cannot use 'skTemp' here cause we really need the copy for the VM :-(
-  var temp = newSym(skVar, getIdent(g.cache, genPrefix), nextId(idgen), owner, n.info, owner.options)
+  let temp =
+    if owner == nil:
```
Comment (Araq) [inline compiler/lowerings.nim]: Owner shoulnd't be nil.

Comment (ringabout) [reply compiler/lowerings.nim]: I will look into this tomorrow.

Comment (ringabout) [reply compiler/lowerings.nim]: `c.prc` != nil but `c.prc.sym` == nil

Comment (ringabout) [reply compiler/lowerings.nim]: I think `c.prc` is initialized in `refresh` proc, so it isn't nil, but `c.prc.sym` is nil.
```nim
proc refresh*(c: PCtx, module: PSym; idgen: IdGenerator) =
  c.module = module
  echo "refresh: ", module
  c.prc = PProc(blocks: @[])
  c.loopIterations = c.config.maxLoopIterationsVM
  c.idgen = idgen
```



```
.refresh: test@369098752
mSwap: test@369098752
```

**Pull request merged**